[Zan Hao] iP

paotheroo opened the pull request:
## DukePro
“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- FAST SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is FREE!

Features:

- [x] Managing tasks
- [x] Managing deadlines
- [x] Managing events
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

paotheroo committed:
Duke: Level-1

paotheroo committed:
Duke: Level-2

paotheroo committed:
Duke Level-3

paotheroo committed:
Duke: Level-4

paotheroo committed:
Duke: A-TextUiTesting

paotheroo committed:
Duke: Level-5

paotheroo committed:
Duke: Level-6

jchilling reviewed (commented):
Looks good! 👍

jhchee18 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,136 @@
+//        Task[] list = new Task[100];
+        ArrayList<Task> list = new ArrayList<>();
+        int i = 0;
+        boolean loop = true;
+        System.out.println("Hello! Duke here, how can I help you?");
+        Scanner sc = new Scanner(System.in);
+
+        while (loop) {
```
I think you should name your boolean variable as `isSomething` to represent it is actually a boolean. You can refer to the naming convention provided in the text book.

jhchee18 commented on src/main/java/DukeException.java:
```diff
@@ -0,0 +1,6 @@
+public class DukeException extends RuntimeException {
+
+    public DukeException(String errorMsg){
+        super(errorMsg);
+    }
+}
```
Should this be extending `Exception` instead of just `RuntimeException`, so that it can catch more general exception?

jhchee18 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,136 @@
+                continue;
+            }
+
+            String str = sc.next();
```
I think the variable `str` can be representative and descriptive like `userInput`.

jhchee18 reviewed (commented):
Overall looks good! 👍 Just some tweaks on the coding standards will be good to go. Good job!

paotheroo committed:
Duke: Level-7

paotheroo committed:
Duke: Level-8

paotheroo committed:
Duke: A-MoreOOP

paotheroo committed:
Duke: A-Packages

paotheroo committed:
Duke: A-Gradle

ChryslineLim commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,271 @@
+//            writer.close();
+//        } catch (IOException e) {}
+//    }
+//}
```
Perhaps you can remove the long chunk of commented out code if it's not being used!

ChryslineLim commented on src/main/java/duke/command/Parser.java:
```diff
@@ -0,0 +1,44 @@
+            try {
+                int number = Integer.parseInt(command);
+                return new UnmarkCommand(number);
+            } catch (NumberFormatException e) {}
```
Perhaps you can print the error in the catch block?

ChryslineLim commented on src/main/java/duke/command/Parser.java:
```diff
@@ -0,0 +1,44 @@
+public class Parser {
+
+    private static final String[] acceptedCommands = { "list", "mark", "unmark", "delete",
+            "todo", "deadline", "event", "bye" };
```
I like that you put the accepted commands into an array!

ChryslineLim commented on src/main/java/duke/command/TaskCommand.java:
```diff
@@ -0,0 +1,53 @@
+
+import duke.Storage;
+import duke.Ui;
+import duke.task.*;
```
It would be better to import each file individually!

ChryslineLim commented on src/main/java/duke/command/UnmarkCommand.java:
```diff
@@ -0,0 +1,30 @@
+    public boolean isExit() {
+        return false;
+    }
+}
```
You can consider combining MarkCommand and UnmarkCommand into a single file with an isDone boolean to represent the state!

ChryslineLim commented on src/main/java/duke/command/UnmarkCommand.java:
```diff
@@ -0,0 +1,30 @@
+            tasks.unmark(index);
+            storage.unmark(index);
+            ui.showMessage("Got it! Task " + index + " has not yet been completed:\n  " + tasks.showTask(index));
+        } catch (IndexOutOfBoundsException e) {
+            ui.showError("You've given me an invalid task to unmark!");
+        }
```
I like that many exceptions are handled for different user input cases!

ChryslineLim commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,52 @@
+package duke.task;
+
```
You might want to standardise across files if a line is left in between package and the first import statement or not!

ChryslineLim reviewed (commented):
Overall, I think that you did great! There are barely any coding standard violations, and I like the way you structured your files into the different packages!

paotheroo committed:
A-JavaDoc

paotheroo committed:
Merge pull request #1 from AY2223S1-CS2103T-F12-2/add-gradle-support

Add Gradle support

paotheroo committed:
Add JUnit tests to test the behavior of the code

paotheroo committed:
Give users a way to find a task by searching for a keyword

paotheroo committed:
Use checkStyle to detect coding style violations

paotheroo committed:
Add a GUI to Duke. Use the JavaFX technology to implement the GUI

paotheroo committed:
Use assert feature

This is to document important assumptions that should hold at various points in the code.

Let's add an assert statement to the logic flow of task command.

Doing so allows us to ensure that the input task type is not faulty.

paotheroo committed:
Critically examines the code and refactor to improve the code quality where necessary

paotheroo committed:
Support the managing of tasks that need to be done after a specific time/task

paotheroo committed:
Add a User Guide to the project

paotheroo committed:
Pre-release updates

paotheroo committed:
Updated User Guide for clarity

paotheroo committed:
Minor bugfixes and code cleanup
